metrics/src/lib.rs

```diff
@@ -163,6 +170,11 @@ impl MetricEntry {
     pub fn name(&self) -> &str {
         &*self.name
     }
+
+    /// Get the description of this metric.
+    pub fn description(&self) -> &str {
```

Comment:
I think I understand why we're representing the description as a `str` even if it's not provided. But should we return an `Option<&str>` here if the `str` is zero-length?

Reply:
yeah that makes sense let me give that a try

Reply:
@brayniac ok so I went with the Option approach but I decided that if either the description is unspecified in the macro annotation, or if the description is a blank string, in both cases the description will default to `None`. I primarily did this so that you can do: 
```rust
#[metric(description = "some metric with a description")]
static METRIC_WITH_DESCRIPTION_NO_NAME: Counter = Counter::new();
```
instead of:
```rust
#[metric(description = Some("some metric with a description"))]
static METRIC_WITH_DESCRIPTION_NO_NAME: Counter = Counter::new();
```
which seems more ergonomic. Not sure there's a better way to achieve that outcome, though

Reply:
Ahh, I was thinking in the description function to do the check for an empty description and return option there, but store it as a str anyway. I think what you did is fine though. Definitely has the ergonomics I was expecting